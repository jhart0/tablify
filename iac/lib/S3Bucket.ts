import { App, Stack, RemovalPolicy } from '@aws-cdk/core'
import { Bucket } from '@aws-cdk/aws-s3'

export class S3Bucket extends Stack {
  readonly bucket: Bucket

  constructor(scope: App, id: string) {
    super(scope, id)
    this.bucket = new Bucket(this, 'site-bucket', {
      bucketName: 'sample-bucket-cdk-tutorial',
      publicReadAccess: true,
      removalPolicy: RemovalPolicy.DESTROY,
      websiteIndexDocument: 'index.html',
    })
  }
}
