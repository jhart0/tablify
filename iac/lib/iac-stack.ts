import * as cloudfront from '@aws-cdk/aws-cloudfront'
import * as route53 from '@aws-cdk/aws-route53'
import * as s3 from '@aws-cdk/aws-s3'
import * as s3deploy from '@aws-cdk/aws-s3-deployment'
import * as acm from '@aws-cdk/aws-certificatemanager'
import * as cdk from '@aws-cdk/core'
import * as targets from '@aws-cdk/aws-route53-targets/lib'

export interface StaticSiteProps {
  domainName: string
  siteSubDomain: string
  staticContentPath: string
}

export class StaticSite extends cdk.Construct {
  constructor(scope: cdk.Stack, id: string, props: StaticSiteProps & cdk.StackProps) {
    super(scope, id)

    const zone = route53.HostedZone.fromLookup(this, 'Zone', { domainName: props.domainName })
    const siteDomain = props.siteSubDomain + '.' + props.domainName
    // tslint:disable-next-line: no-unused-expression
    new cdk.CfnOutput(this, 'Site', { value: 'https://' + siteDomain })

    const siteBucket = new s3.Bucket(this, 'tablify-bucket', {
      bucketName: 'tablify-static',
      websiteIndexDocument: 'index.html',
      websiteErrorDocument: 'error.html',
      publicReadAccess: true,
    })
    // tslint:disable-next-line: no-unused-expression
    new cdk.CfnOutput(this, 'Bucket', { value: siteBucket.bucketName })

    const certificateArn = new acm.DnsValidatedCertificate(this, 'tablify-cert', {
      domainName: props.domainName,
      hostedZone: zone,
      region: 'us-east-1',
      subjectAlternativeNames: ['*.' + props.domainName],
    }).certificateArn
    // tslint:disable-next-line: no-unused-expression
    new cdk.CfnOutput(this, 'Certificate', { value: certificateArn })

    const distribution = new cloudfront.CloudFrontWebDistribution(
      this,
      'tablify-distribution',
      {
        aliasConfiguration: {
          acmCertRef: certificateArn,
          names: [props.domainName, siteDomain],
          sslMethod: cloudfront.SSLMethod.SNI,
          securityPolicy: cloudfront.SecurityPolicyProtocol.TLS_V1_1_2016,
        },
        originConfigs: [
          {
            s3OriginSource: {
              s3BucketSource: siteBucket,
            },
            behaviors: [{ isDefaultBehavior: true }],
          },
        ],
      }
    )
    // tslint:disable-next-line: no-unused-expression
    new cdk.CfnOutput(this, 'DistributionId', { value: distribution.distributionId })

    // tslint:disable-next-line: no-unused-expression
    new route53.ARecord(this, 'tablify-dns-apex', {
      recordName: siteDomain,
      target: route53.RecordTarget.fromAlias(new targets.CloudFrontTarget(distribution)),
      zone,
    })

    // tslint:disable-next-line: no-unused-expression
    new route53.ARecord(this, 'tablify-dns', {
      recordName: props.domainName,
      target: route53.RecordTarget.fromAlias(new targets.CloudFrontTarget(distribution)),
      zone,
    })

    // tslint:disable-next-line: no-unused-expression
    new s3deploy.BucketDeployment(this, 'tablify-deployment', {
      sources: [s3deploy.Source.asset(props.staticContentPath)],
      destinationBucket: siteBucket,
      distribution,
      distributionPaths: ['/*'],
    })
  }
}
