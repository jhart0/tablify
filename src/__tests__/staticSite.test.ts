import * as cdk from '@aws-cdk/core'
import * as staticSite from '../../iac/lib/iac-stack'
import '@aws-cdk/assert/jest'

describe('IaC Tests', () => {
  const app = new cdk.App()
  const stack = new cdk.Stack(app, 'testStack', {
    env: { account: '1234567890', region: 'dummy' },
  })
  // tslint:disable-next-line: no-unused-expression
  new staticSite.StaticSite(stack, 'testStaticSite', {
    siteSubDomain: 'www',
    domainName: 'test.com',
    staticContentPath: './build',
  })

  test('static site creates bucket', () => {
    expect(stack).toHaveResource('AWS::S3::Bucket', {
      BucketName: 'www.test.com',
      WebsiteConfiguration: {
        IndexDocument: 'index.html',
        ErrorDocument: 'error.html',
      },
    })
  })

  test('static site creates cloudfront distribution', () => {
    expect(stack).toHaveResource('AWS::CloudFront::Distribution', {
      DistributionConfig: {
        Aliases: ['test.com', 'www.test.com'],
        DefaultCacheBehavior: {
          AllowedMethods: ['GET', 'HEAD'],
          CachedMethods: ['GET', 'HEAD'],
          Compress: true,
          ForwardedValues: {
            Cookies: {
              Forward: 'none',
            },
            QueryString: false,
          },
          TargetOriginId: 'origin1',
          ViewerProtocolPolicy: 'redirect-to-https',
        },
        DefaultRootObject: 'index.html',
        Enabled: true,
        HttpVersion: 'http2',
        IPV6Enabled: true,
        Origins: [
          {
            ConnectionAttempts: 3,
            ConnectionTimeout: 10,
            DomainName: {
              'Fn::GetAtt': ['testStaticSitetablifyprodbucketEBCA2C99', 'RegionalDomainName'],
            },
            Id: `origin1`,
            S3OriginConfig: {},
          },
        ],
        PriceClass: `PriceClass_100`,
        ViewerCertificate: {
          AcmCertificateArn: {
            'Fn::GetAtt': [
              'testStaticSitetablifyprodcertCertificateRequestorResource3A3455F6',
              'Arn',
            ],
          },
          MinimumProtocolVersion: `TLSv1.1_2016`,
          SslSupportMethod: `sni-only`,
        },
      },
    })
  })
})
