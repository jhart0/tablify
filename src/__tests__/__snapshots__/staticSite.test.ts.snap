// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`static site creates bucket 1`] = `
Object {
  "Outputs": Object {
    "testStaticSite3E4991B7": Object {
      "Value": "https://www.test.com",
    },
    "testStaticSiteBucket07DA531C": Object {
      "Value": Object {
        "Ref": "testStaticSitetablifyprodbucketEBCA2C99",
      },
    },
    "testStaticSiteCertificate226A8652": Object {
      "Value": Object {
        "Fn::GetAtt": Array [
          "testStaticSitetablifyprodcertCertificateRequestorResource3A3455F6",
          "Arn",
        ],
      },
    },
    "testStaticSiteDistributionId9F129477": Object {
      "Value": Object {
        "Ref": "testStaticSitetablifyproddistributionCFDistribution9A8A7215",
      },
    },
  },
  "Parameters": Object {
    "AssetParameters19e461d2ff1a5b90438fed6ceee4c197d7efee8712a6f76d85b501ab20bfb1a2ArtifactHash652C125C": Object {
      "Description": "Artifact hash for asset \\"19e461d2ff1a5b90438fed6ceee4c197d7efee8712a6f76d85b501ab20bfb1a2\\"",
      "Type": "String",
    },
    "AssetParameters19e461d2ff1a5b90438fed6ceee4c197d7efee8712a6f76d85b501ab20bfb1a2S3BucketFCCD3A76": Object {
      "Description": "S3 bucket for asset \\"19e461d2ff1a5b90438fed6ceee4c197d7efee8712a6f76d85b501ab20bfb1a2\\"",
      "Type": "String",
    },
    "AssetParameters19e461d2ff1a5b90438fed6ceee4c197d7efee8712a6f76d85b501ab20bfb1a2S3VersionKey07AF06B6": Object {
      "Description": "S3 key for asset version \\"19e461d2ff1a5b90438fed6ceee4c197d7efee8712a6f76d85b501ab20bfb1a2\\"",
      "Type": "String",
    },
    "AssetParameters4184245adc1f2ed71e1f0ae5719f8fd7f34324b750f1bf06b2fb5cf1f4014f50ArtifactHash846130E4": Object {
      "Description": "Artifact hash for asset \\"4184245adc1f2ed71e1f0ae5719f8fd7f34324b750f1bf06b2fb5cf1f4014f50\\"",
      "Type": "String",
    },
    "AssetParameters4184245adc1f2ed71e1f0ae5719f8fd7f34324b750f1bf06b2fb5cf1f4014f50S3BucketB6159468": Object {
      "Description": "S3 bucket for asset \\"4184245adc1f2ed71e1f0ae5719f8fd7f34324b750f1bf06b2fb5cf1f4014f50\\"",
      "Type": "String",
    },
    "AssetParameters4184245adc1f2ed71e1f0ae5719f8fd7f34324b750f1bf06b2fb5cf1f4014f50S3VersionKey2060CAC0": Object {
      "Description": "S3 key for asset version \\"4184245adc1f2ed71e1f0ae5719f8fd7f34324b750f1bf06b2fb5cf1f4014f50\\"",
      "Type": "String",
    },
    "AssetParametersb103d049d9e8f1b490264222976241c73b1f20519fb69b814a999d8636fb02f1ArtifactHash8A3F0E30": Object {
      "Description": "Artifact hash for asset \\"b103d049d9e8f1b490264222976241c73b1f20519fb69b814a999d8636fb02f1\\"",
      "Type": "String",
    },
    "AssetParametersb103d049d9e8f1b490264222976241c73b1f20519fb69b814a999d8636fb02f1S3Bucket4593222B": Object {
      "Description": "S3 bucket for asset \\"b103d049d9e8f1b490264222976241c73b1f20519fb69b814a999d8636fb02f1\\"",
      "Type": "String",
    },
    "AssetParametersb103d049d9e8f1b490264222976241c73b1f20519fb69b814a999d8636fb02f1S3VersionKey3728D0D2": Object {
      "Description": "S3 key for asset version \\"b103d049d9e8f1b490264222976241c73b1f20519fb69b814a999d8636fb02f1\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536": Object {
      "DependsOn": Array [
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters4184245adc1f2ed71e1f0ae5719f8fd7f34324b750f1bf06b2fb5cf1f4014f50S3BucketB6159468",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters4184245adc1f2ed71e1f0ae5719f8fd7f34324b750f1bf06b2fb5cf1f4014f50S3VersionKey2060CAC0",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters4184245adc1f2ed71e1f0ae5719f8fd7f34324b750f1bf06b2fb5cf1f4014f50S3VersionKey2060CAC0",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
            "Arn",
          ],
        },
        "Runtime": "python3.6",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParametersb103d049d9e8f1b490264222976241c73b1f20519fb69b814a999d8636fb02f1S3Bucket4593222B",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParametersb103d049d9e8f1b490264222976241c73b1f20519fb69b814a999d8636fb02f1S3Bucket4593222B",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "testStaticSitetablifyprodbucketEBCA2C99",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "testStaticSitetablifyprodbucketEBCA2C99",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "cloudfront:GetInvalidation",
                "cloudfront:CreateInvalidation",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF",
        "Roles": Array [
          Object {
            "Ref": "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "testStaticSitetablifyprodbucketEBCA2C99": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketName": "www.test.com",
        "WebsiteConfiguration": Object {
          "ErrorDocument": "error.html",
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "testStaticSitetablifyprodbucketPolicy9ADB050C": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "testStaticSitetablifyprodbucketEBCA2C99",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": "*",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "testStaticSitetablifyprodbucketEBCA2C99",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "testStaticSitetablifyprodcertCertificateRequestorFunctionEAD6B9FF": Object {
      "DependsOn": Array [
        "testStaticSitetablifyprodcertCertificateRequestorFunctionServiceRoleDefaultPolicy074BFEC8",
        "testStaticSitetablifyprodcertCertificateRequestorFunctionServiceRole4F98BC1E",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters19e461d2ff1a5b90438fed6ceee4c197d7efee8712a6f76d85b501ab20bfb1a2S3BucketFCCD3A76",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters19e461d2ff1a5b90438fed6ceee4c197d7efee8712a6f76d85b501ab20bfb1a2S3VersionKey07AF06B6",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters19e461d2ff1a5b90438fed6ceee4c197d7efee8712a6f76d85b501ab20bfb1a2S3VersionKey07AF06B6",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.certificateRequestHandler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "testStaticSitetablifyprodcertCertificateRequestorFunctionServiceRole4F98BC1E",
            "Arn",
          ],
        },
        "Runtime": "nodejs10.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "testStaticSitetablifyprodcertCertificateRequestorFunctionServiceRole4F98BC1E": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "testStaticSitetablifyprodcertCertificateRequestorFunctionServiceRoleDefaultPolicy074BFEC8": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "acm:RequestCertificate",
                "acm:DescribeCertificate",
                "acm:DeleteCertificate",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": "route53:GetChange",
              "Effect": "Allow",
              "Resource": "*",
            },
            Object {
              "Action": "route53:changeResourceRecordSets",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":route53:::hostedzone/DUMMY",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "testStaticSitetablifyprodcertCertificateRequestorFunctionServiceRoleDefaultPolicy074BFEC8",
        "Roles": Array [
          Object {
            "Ref": "testStaticSitetablifyprodcertCertificateRequestorFunctionServiceRole4F98BC1E",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "testStaticSitetablifyprodcertCertificateRequestorResource3A3455F6": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DomainName": "www.test.com",
        "HostedZoneId": "DUMMY",
        "Region": "us-east-1",
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "testStaticSitetablifyprodcertCertificateRequestorFunctionEAD6B9FF",
            "Arn",
          ],
        },
      },
      "Type": "AWS::CloudFormation::CustomResource",
      "UpdateReplacePolicy": "Delete",
    },
    "testStaticSitetablifyproddeploymentCustomResourceC4BC15CF": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DestinationBucketName": Object {
          "Ref": "testStaticSitetablifyprodbucketEBCA2C99",
        },
        "DistributionId": Object {
          "Ref": "testStaticSitetablifyproddistributionCFDistribution9A8A7215",
        },
        "DistributionPaths": Array [
          "/*",
        ],
        "Prune": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536",
            "Arn",
          ],
        },
        "SourceBucketNames": Array [
          Object {
            "Ref": "AssetParametersb103d049d9e8f1b490264222976241c73b1f20519fb69b814a999d8636fb02f1S3Bucket4593222B",
          },
        ],
        "SourceObjectKeys": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersb103d049d9e8f1b490264222976241c73b1f20519fb69b814a999d8636fb02f1S3VersionKey3728D0D2",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersb103d049d9e8f1b490264222976241c73b1f20519fb69b814a999d8636fb02f1S3VersionKey3728D0D2",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        ],
      },
      "Type": "Custom::CDKBucketDeployment",
      "UpdateReplacePolicy": "Delete",
    },
    "testStaticSitetablifyproddistributionCFDistribution9A8A7215": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "Aliases": Array [
            "www.test.com",
          ],
          "DefaultCacheBehavior": Object {
            "AllowedMethods": Array [
              "GET",
              "HEAD",
            ],
            "CachedMethods": Array [
              "GET",
              "HEAD",
            ],
            "Compress": true,
            "ForwardedValues": Object {
              "Cookies": Object {
                "Forward": "none",
              },
              "QueryString": false,
            },
            "TargetOriginId": "origin1",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "ConnectionAttempts": 3,
              "ConnectionTimeout": 10,
              "DomainName": Object {
                "Fn::GetAtt": Array [
                  "testStaticSitetablifyprodbucketEBCA2C99",
                  "RegionalDomainName",
                ],
              },
              "Id": "origin1",
              "S3OriginConfig": Object {},
            },
          ],
          "PriceClass": "PriceClass_100",
          "ViewerCertificate": Object {
            "AcmCertificateArn": Object {
              "Fn::GetAtt": Array [
                "testStaticSitetablifyprodcertCertificateRequestorResource3A3455F6",
                "Arn",
              ],
            },
            "MinimumProtocolVersion": "TLSv1.1_2016",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "testStaticSitetablifyproddns858491E9": Object {
      "Properties": Object {
        "AliasTarget": Object {
          "DNSName": Object {
            "Fn::GetAtt": Array [
              "testStaticSitetablifyproddistributionCFDistribution9A8A7215",
              "DomainName",
            ],
          },
          "HostedZoneId": "Z2FDTNDATAQYW2",
        },
        "HostedZoneId": "DUMMY",
        "Name": "www.test.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
  },
}
`;
